lib/HLSL/DxilValidation.cpp

```diff
@@ -1429,7 +1401,10 @@ static unsigned StoreValueToMask(ArrayRef<Value *> vals) {
 
 static int GetCBufSize(Value *cbHandle, ValidationContext &ValCtx) {
   if (!isa<CallInst>(cbHandle)) {
-    ValCtx.EmitError(ValidationRule::InstrHandleNotFromCreateHandle);
+    if (Instruction *I = dyn_cast<Instruction>(cbHandle))
```

Comment:
Just a comment..  It seems that this handle validation code is repeated several times, and could be wrapped in a function.  Also, it doesn't verify that the function it comes from is the correct intrinsic.  Adding validation would benefit from this being wrapped in a function as well.

Reply:
Forgot to remove the confused section here as it was clarified offline.